src/entities/Proposal/jobs.ts

```diff
@@ -1,168 +1,169 @@
-    const { winnerChoice, winnerVotingPower } = calculateWinnerChoice(results)
+async function updateFinishedProposals(finishedProposals: ProposalWithOutcome[], context: JobContext) {
+  if (finishedProposals.length > 0) {
+    context.log(`Finishing ${finishedProposals.length} proposals...`)
+    await ProposalModel.finishProposal(
+      finishedProposals.map(({ id }) => id),
+      ProposalStatus.Finished
+    )
+  }
+}
 
-    const outcome: Outcome = {
-      winnerChoice,
-    }
+function getBudgetForProposal(currentBudgets: CurrentBudget[], proposal: ProposalWithOutcome) {
+  return currentBudgets.find((budget) => budget.start_at <= proposal.start_at && budget.finish_at > proposal.start_at)
+}
 
-    const invalidOption = INVALID_PROPOSAL_POLL_OPTIONS.toLocaleLowerCase()
-    const isYesNo = sameOptions(choices, ['yes', 'no'])
-    const isAcceptReject = sameOptions(choices, ['accept', 'reject', invalidOption])
-    const isForAgainst = sameOptions(choices, ['for', 'against', invalidOption])
-
-    const minimumVotingPowerRequired = proposal.required_to_pass || 0
-    if (winnerVotingPower === 0 || winnerVotingPower < minimumVotingPowerRequired) {
-      outcome.outcomeStatus = ProposalOutcome.REJECTED
-    } else if (winnerChoice === invalidOption) {
-      outcome.outcomeStatus = ProposalOutcome.REJECTED
-    } else if (isYesNo || isAcceptReject || isForAgainst) {
-      if (
-        (isYesNo && results['yes'] > results['no']) ||
-        (isAcceptReject && results['accept'] > results['reject']) ||
-        (isForAgainst && results['for'] > results['against'])
-      ) {
-        outcome.outcomeStatus = ProposalOutcome.ACCEPTED
-      } else {
-        outcome.outcomeStatus = ProposalOutcome.REJECTED
-      }
-    } else {
-      outcome.outcomeStatus = ProposalOutcome.FINISHED
-    }
+function getCategoryBudgetFor(proposal: ProposalWithOutcome, proposalBudget: CurrentBudget) {
+  const categoryName = snakeCase(proposal.configuration.category)
+  return proposalBudget.categories[categoryName]
+}
 
-    return outcome
-  } catch (e) {
-    context.error(`Unable to calculate outcome for ${proposal.snapshot_id}`, e as Error)
-  }
+function grantCanBeFunded(proposal: ProposalWithOutcome, proposalBudget: CurrentBudget) {
+  const categoryBudget = getCategoryBudgetFor(proposal, proposalBudget)
+  const size = asNumber(proposal.configuration.size)
+  return categoryBudget.allocated + size <= categoryBudget.total
 }
 
-export async function finishProposal(context: JobContext) {
-  const pendingProposals = await ProposalModel.getFinishedProposals()
-  if (pendingProposals.length === 0) {
-    context.log(`No finished proposals...`)
-    return
-  }
+function updateCategoryBudget(proposal: ProposalWithOutcome, proposalBudget: CurrentBudget) {
+  const categoryBudget = getCategoryBudgetFor(proposal, proposalBudget)
+  const size = asNumber(proposal.configuration.size)
+  categoryBudget.allocated += size
+  categoryBudget.available -= size
```

Comment:
I think these two are not necessary as you are not using categoryBudget anymore 🤔